### DISCUSSION TITLE
Skip payment details if Store credit covers the price of the product

### DISCUSSION THREAD

**Discussion by diegomichel** [solidusio/solidus#4725] [Troubleshooting]
Hello,

I wonder if there is a way to skip asking for credit card details(using solidus_stripe) if the store credit covers the price of a product. Currently, it asks for the credit card details even if it doesn't use them in the end.

I tried enabling it for users on admin, but it throws an error about missing a template: https://github.com/solidusio/solidus/issues/2848

Thanks!.

**Comment by waiting-for-dev** [comment]
Not sure if somebody can provide a better idea, but having a quick look, it seems you could override the [`#create_intent` action](https://github.com/solidusio/solidus_stripe/blob/dc42bfe114f817937710c697ff0760eb088ff13c/app/controllers/solidus_stripe/intents_controller.rb#L7) and check for your condition. Take a look here for how to do it in the less dangerous way: https://guides.solidus.io/customization/customizing-the-core#using-overrides
